### PR TITLE
edit metrics doc with telemetry requirements

### PR DESCRIPTION
**Pull request by irrationalagent**
this provides some specs for https://github.com/mozilla/fxa-discoverability-shield-study/issues/7

see the [revised doc rendered in markdown](https://github.com/mozilla/fxa-discoverability-shield-study/blob/metrics-docs/docs/TELEMETRY.md)

OK, might be worth a short meeting soon work out some details here.

1. The way shield is designed its easiest to send one ping per event, which seems inefficient to me, but its best we follow that precedent. 
2. We want to use the `shield-study-addon` ping schema for this, example [here](https://github.com/motin/taar-experiment-v3-shield-study/blob/develop/schemas/full-schema.json).
3. I've defined three ping "types" for enrollment, unenrollment and UI engagement events, see doc
4. I wonder if we could get away with sending the FxA user ID, if signed in, in this ping. Can we obtain it from the browser or during authentication? It could potentially allow us to join the experiment data on fxa flow events. At the very least, we should define params so that login and reg flows originating from the addon can be identified by their branch in the experiment (though I know this may not be possible for control). This would enable join-ability of client telemetry and fxa data for users enrolled in the experiment, which we may need to run past higher-ups as it has been a previous no-no. 
5. I don't want to slow down the deployment process too much, but I'm slightly worried that the addition of the toolbar icon will interfere with people trying to access the hamburger menu. Unfortunately I'm not seeing any existing browser telemetry around events relating to hamburger menu clicks. Maybe its OK to move forward with this initial version but we should look into landing code in tree that will allow this as a fast follow up - it shouldn't be too hard. Also I don't think there is any browser telemetry around views/interactions with sync prefs :(. This hampers our ability to measure the behavior of the control branch. As far as I can tell, for control, all we will be able to measure is whether sync/fxa are configured.

### CODE REVIEW

File: docs/TELEMETRY.md [hunk 1]
```diff
@@ -23,35 +23,75 @@
...
+        "fxaState": {
+          "type": "string" // one of: verfied, unverfied, none
+        },
+        "interactionType": {
+          "type": "string" // see list above for possible values, or `none`
+        },
+        "doorhangerActiveSeconds": {
+          "type": "string" // number of seconds doorhanger was displayed for (0 if none)
+        },
+        "hasAvatar": {
```
Comment (vbudhram) [inline docs/TELEMETRY.md]: > I wonder if we could get away with sending the FxA user ID

We do have access to the UID if we decided to send it. Below are some of the other values that we have access to

```javascript
{
    email: "vbudhram+test@mozilla.com",
    locale: "en-US,en;q=0.5",
    amrValues: [],
    twoFactorAuthentication: false,
    uid: "5298b1da5fb64496a8a94a8c141ff99f",
    avatar: "https://firefoxusercontent.com/c3f000621a1717c9c29eb606590f8e41",
    avatarDefault: false,
    displayName: "sdfdd",
    verified: true
}
```

Not sure if `displayName` and `twoFactorAuthentication` would have any value here.

File: docs/TELEMETRY.md [hunk 2]
```diff
@@ -23,35 +23,75 @@
 
 The STUDY SPECIFIC ENDINGS this study supports are:
 
+I don't think there are any ways this study can end besides being manually disabled or unenrolled? I think survey should be shown to all enrollees at unenrollment.
+
```
Comment (vbudhram) [inline docs/TELEMETRY.md]: Should we make a section to define/describe the treatment and control groups?

**Comment by philbooth** [comment]
> > I wonder if we could get away with sending the FxA user ID
>
> We do have access to the UID if we decided to send it.

Firstly, we definitely can't send that uid to telemetry because it could be used to look up PII in the db. Nor can we send some of the other details you list there, like `email`, `avatar` or `displayName`. Those are all PII.

> It could potentially allow us to join the experiment data on fxa flow events. 

The uid in the flow events is hashed, but unfortunately we can't even hash the uid here to match it. The HMAC key that we hash it with in the back-end is considered a production secret (again, because PII), there's no way we could include it in client-side code.

The client-side Sync code does have a copy of the hashed uid though, in the response from the token server, so maybe we could pull it out from there. Alternatively, we could include it in the response from the FxA back-end and then it expose it through `fxAccounts::getSignedInUser` with the rest of the stuff we have there.

Which reminds me: #12

**Comment by shane-tomlinson** [comment]
> Firstly, we definitely can't send that uid to telemetry because it could be used to look up PII in the db. 
> The uid in the flow events is hashed, but unfortunately we can't even hash the uid here to match it. 

There is a plan for the Telemetry team to ingest the FxA logs with the raw uid and hash it themselves with the same HMAC key we use. @irrationalagent, do you happen to know how that work is progressing?

File: docs/TELEMETRY.md (hunk 2 above)
Comment (irrationalagent) [reply docs/TELEMETRY.md]: Good point, I'll add that

**Comment by irrationalagent** [comment]
> There is a plan for the Telemetry team to ingest the FxA logs with the raw uid and hash it themselves with the same HMAC key we use. @irrationalagent, do you happen to know how that work is progressing?

Not fast enough to help us here.

If we can obtain the hashed uid from the browser without too much trouble, then let's try that. Making changes to a server endpoint to expose it from there sounds like too much of a change to be worth it.

**Comment by philbooth** [comment]
> If we can obtain the hashed uid from the browser without too much trouble, then let's try that.

See #22.

It's not perfect, because it's not available immediately after signing in, so won't be present on some events. But it's better than nothing, I guess?

**Comment by irrationalagent** [comment]
This is awesome, much better than nothing, thanks Phil.

**Comment by vbudhram** [comment]
@irrationalagent What are your thoughts on the name for this entrypoint, https://github.com/mozilla/fxa-discoverability-shield-study/blob/6b2b7011d8f8823adcfea6ecc5abac87472599a8/src/popup/fxa.js#L3? Should we document that in here?

**Comment by irrationalagent** [comment]
@vbudhram I think we should include the version of the addon somewhere, either appended to the entrypoint itself or in another parameter. Potentially the cohort as well, although I realize it will only be applied to treatment (just planning ahead for future iterations)

**Comment by vbudhram** [comment]
@irrationalagent, ah missed your comment. I'll verify that version information is append in telemetry data.

**Comment by vbudhram** [comment]
> I'll verify that version information is append in telemetry data.

Confirmed! Extension version is sent on every telemetry ping.

**Pull request merged**